bsp/Src/CAN.c

```diff
@@ -666,8 +669,17 @@ void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* hcan) {
     if (hcan->Instance == CAN1) {
       for (int i = 0; i < can1_recv_entry_count; i++) {
         if (can1_recv_entries[i].id == payload.header.StdId) {
-          xQueueSendFromISR(can1_recv_entries[i].queue, &payload,
+          if (can1_recv_entries[i].circular){
+            xQueueSendCircularBufferFromISR(
+              can1_recv_entries[i].queue, 
+              &payload, 
+              &higherPriorityTaskWoken, 
+              sizeof(rx_payload_t)
+            );
+          } else {
+            xQueueSendFromISR(can1_recv_entries[i].queue, &payload,
                             &higherPriorityTaskWoken);
+          }
```

Comment:
This might be kinda silly or better. But let's make it oop-y. Have a function pointer to the correct function so you don't have to have the if statement here.

Reply:
💀💀💀 maybe

Reply:
Since SendCircularBufferFromISR and SendFromISR are both macros, I don't think we can actually do this 💀

Reply:
Just make them not macros

Reply:
SendFromISR is implemented by FreeRTOS and I'd rather not touch that; I could make SendCircBufFromISR not a macro but honestly the value of the function pointer here is not super clear to me. Is it just to avoid one branch instruction in the interrupt?

Reply:
Yes. I am just bringing up the idea. Merge if you want already (if tested, which I think it was)